net/graphsync_fetcher.go

```diff
@@ -24,10 +24,10 @@ import (
 var logGraphsyncFetcher = logging.Logger("net.graphsync_fetcher")
 
 const (
-	// Timeout for a single graphsync request (which may be for many blocks).
-	// We might prefer this timeout to scale with the number of blocks expected in the fetch,
-	// when that number is large.
-	requestTimeout = 60 * time.Second
+	// Timeout for a single graphsync request getting "stuck"
+	// -- if no more responses are received for a period greater than this,
+	// we will assume the request has hung-up and cancel it
+	unresponsiveTimeout = 10 * time.Second
```

Comment:
I'm curious -- do we have information on the upper bound of the delay we would expect with high probability from a peer with no network issues?  My intuition is that we want to set this as low as we can reasonably get away with before we start killing productive connections.  My uninformed intuition is also that 10 seconds is probably higher than we need and I'd love to know if this is wrong and 10 seconds is already pushing the limit.

Reply:
The short version is I honestly don't know :( I would probably prefer to wait on dropping it lower till we are at least requesting two requests in parallel. Then if we get a false positive on believing ourselves unresponsive we can at least still pickup from the other request.

Reply:
another option would be to track actual latency and use a multiple of the average latency between progress steps

Reply:
`defaultProgressTimeout`

Reply:
> another option would be to track actual latency

Not sure how it fits into priorities and I'm guessing this comes after other fires are put out but I love the idea of making this decision using observations.   I'm imagining gathering a distribution of latencies over different data (single block, long chains) separating into "healthy" and "unhealthy" connections and taking the cutoff point to be some threshold (95% of latencies grouped into "healthy connections" for example).